### PR TITLE
Added MigrateUsersToPro.ps1 and updated README with more examples

### PR DESCRIPTION
**Pull request by CodeCyclone**
* Updated README with more examples
* Added MigrateUsersToPro.ps1 in examples folder

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,26 +57,34 @@ Get-Module MicrosoftPowerBIMgmt* -ListAvailable | Uninstall-Module -Force
...
 
-Get workspaces for the user. By default (without -First parameter) it shows the first 100 workspaces assigned to the user.:
+Get workspaces for the user. By default (without `-First` parameter) it shows the first 100 workspaces assigned to the user:
 
 ```powershell
 Get-PowerBIWorkspace
 ```
 
-Introduced -All parameter to show all workspaces assigned to the user.
+With the `-All` parameter, all workspaces assigned to the user will be shown:
```
Comment (nandemoii) [inline README.md]: "Use the `-All` parameter to show all workspaces assigned to the user"

File: README.md [hunk 2]
```diff
@@ -57,26 +57,34 @@ Get-Module MicrosoftPowerBIMgmt* -ListAvailable | Uninstall-Module -Force
...
 
 ```powershell
-Connect-PowerBIServiceAccount   # or Login-PowerBIServiceAccount
+Connect-PowerBIServiceAccount   # or use aliases: Login-PowerBIServiceAccount, Login-PowerBI
 ```
 
 ### Get Workspaces
 
-Get workspaces for the user. By default (without -First parameter) it shows the first 100 workspaces assigned to the user.:
+Get workspaces for the user. By default (without `-First` parameter) it shows the first 100 workspaces assigned to the user:
```
Comment (nandemoii) [inline README.md]: "(i.e. without `-First` parameter)"

File: README.md [hunk 3]
```diff
@@ -103,12 +111,32 @@ Remove-PowerBIWorkspaceUser -Scope Organization -Id 3244f1c1-01cf-457f-9383-6035
...
+Get-PowerBIWorkspace -Scope Organization -Deleted -All
+```
+
 Restores a deleted workspace:
 
 ```powershell
 Restore-PowerBIWorkspace -Id "3244f1c1-01cf-457f-9383-6035e4950fdc" -RestoredName "TestWorkspace" -AdminEmailAddress "john@contoso.com"
 ```
 
+### Repair orphaned workspaces
```
Comment (nandemoii) [inline README.md]: "Recover orphaned workspaces"

File: README.md [hunk 4]
```diff
@@ -103,12 +111,32 @@ Remove-PowerBIWorkspaceUser -Scope Organization -Id 3244f1c1-01cf-457f-9383-6035
...
+
+Workspaces can become orphaned when no administrators are assigned to the workspace, to view you must be a tenant administrator:
```
Comment (nandemoii) [inline README.md]: "A workspace becomes orphaned when it has no assigned administrators. If you are a tenant administrator, run the following to view all orphaned workspaces:"

File: README.md [hunk 5]
```diff
@@ -117,9 +145,121 @@ Get all reports for the user:
...
+
+```powershell
+Get-PowerBIImport
+```
+
+### Create Report
+
+Create a report in Power BI by uploading a *.pbix file:
+
+```powershell
```
Comment (nandemoii) [inline README.md]: Do you know why the text from here appears as italic in this review? The rendered markdown file looks fine.

File: README.md [hunk 6]
```diff
@@ -117,9 +145,121 @@ Get all reports for the user:
...
+
+Get Power BI tables contained within a dataset:
+
+```powershell
+Get-PowerBITable -DatasetId 65d7d7e5-8af0-4e94-b20b-50a882ae15e1
+```
+
+### Call the Power BI Rest API
+
+As the cmdlets haven't been built to handle all the scenarios the [Power BI API](https://docs.microsoft.com/en-us/rest/api/power-bi/) supports, you can reuse the authenticated session from `Connect-PowerBIServiceAccount` to make custom REST requests:
```
Comment (nandemoii) [inline README.md]: "For [Power BI APIs](https://docs.microsoft.com/en-us/rest/api/power-bi/) that lack corresponding cmdlets, you can reuse the authenticated session from `Connect-PowerBIServiceAccount` to make custom REST requests"

File: examples/MigrateUsersToPro.ps1 [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+# Detect if all Pro licenses are consumed
+if($proSku.ActiveUnits -le $proSku.ConsumedUnits) {
+    Write-Warning "You are currently consuming all active units of your Power BI Pro license, please remove users from licenses and try again"
+    Write-Warning $proSku
+    return
+}
+
+# Indicate some users can't be migrated due to lack of Pro licenses
+$availableProLicenses = $proSku.ActiveUnits - $proSku.ConsumedUnits
+Write-Host "Available pro licenses: $availableProLicenses"
```
Comment (nandemoii) [inline examples/MigrateUsersToPro.ps1]: "Number of available pro licenses: $availableProLicenses"

File: examples/MigrateUsersToPro.ps1 [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+    else {
+        Write-Error "User with PUID '' was not migrated to pro because available licenses was zero"
+        $failedToMigrate += $isStandard
+    }
+}
+
+Write-Host "Finished migrating users to pro licenses"
+
+if($usersMigrated.Count -gt 0) {
+    Write-Host "Users migrated to pro, count: $($usersMigrated.Count)" -ForegroundColor Green
```
Comment (nandemoii) [inline examples/MigrateUsersToPro.ps1]: "$($usersMigrated.Count) users migrated to pro licenses"

File: examples/MigrateUsersToPro.ps1 [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+        Write-Host "`nUsers migrated to pro written to: $migratedUserFilePath"
+        $usersMigrated | Export-Csv -Path $migratedUserFilePath -Force -Append -NoTypeInformation -ErrorAction Continue
+    }
+}
+else {
+    Write-Warning "Summary: No users were migrated to pro licenses"
+}
+
+if($alreadyPro.Count -gt 0) {
+    Write-Warning "Users already pro licenses, count: $($alreadyPro.Count)"
```
Comment (nandemoii) [inline examples/MigrateUsersToPro.ps1]: "$($alreadyPro.Count) users already had pro licenses"

File: examples/MigrateUsersToPro.ps1 [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+    
+    if(!$NoResultFile) {
+        # Append for multiple runs
+        $alreadyProFilePath = Join-Path -Path $csvDirectory -ChildPath 'alreadyProUsers.csv'
+        Write-Host "`nAlready pro users written to: $alreadyProFilePath"
+        $alreadyPro | Export-Csv -Path $alreadyProFilePath -Force -Append -NoTypeInformation -ErrorAction Continue
+    }
+}
+else {
+    Write-Host "Summary: No users that were previously pro" -ForegroundColor Green
```
Comment (nandemoii) [inline examples/MigrateUsersToPro.ps1]: "Summary: No users found that already had pro licenses"

File: examples/MigrateUsersToPro.ps1 [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+if($failedToMigrate.Count -gt 0) {
+    Write-Error "Users failed to migrate, count: $($failedToMigrate.Count)"
```
Comment (nandemoii) [inline examples/MigrateUsersToPro.ps1]: "Failed to migrate $($failedToMigrate.Count) users to pro licenses"

File: examples/MigrateUsersToPro.ps1 [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+param
+(
+    [Parameter(Mandatory)]
+    $Path,
+
+    [ValidateNotNullOrEmpty()]
+    [string] $PUIDColumnName = 'Puid',
+
+    [ValidateNotNullOrEmpty()]
+    [string] $LicenseTypeScan = 'POWER_BI_STANDARD',
```
Comment (nandemoii) [inline examples/MigrateUsersToPro.ps1]: This looks like it's unused.

File: examples/MigrateUsersToPro.ps1 [hunk 13]
```diff
@@ -0,0 +1,184 @@
+﻿<#
+.Synopsis
+    Migrates users to Pro Power BI licenses.
```
Comment (nandemoii) [inline examples/MigrateUsersToPro.ps1]: Should there be a script for migrating users to standard licenses? Is this operation easy to undo?

File: README.md (hunk 1 above)
Comment (CodeCyclone) [reply README.md]: Fixed

File: README.md (hunk 5 above)
Comment (CodeCyclone) [reply README.md]: It's because of the *, I will need to escape

File: examples/MigrateUsersToPro.ps1 (hunk 12 above)
Comment (CodeCyclone) [reply examples/MigrateUsersToPro.ps1]: Good catch, I was able to remove a few things thanks to that

File: examples/MigrateUsersToPro.ps1 (hunk 13 above)
Comment (CodeCyclone) [reply examples/MigrateUsersToPro.ps1]: Actually anyone that isn't pro is a standard\free license but there is a specific license type for standard but I will consider anyone not pro a standard user and I removed the one switch

**Pull request merged**